### PR TITLE
[v2.7] Run integration and unit tests in GitHub Actions

### PR DESCRIPTION
**Pull request by rafaelbreno**
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: tests/v2/integration/catalogv2/rancher_managed_charts_test.go [hunk 1]
```diff
@@ -0,0 +1,538 @@
+package integration
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
```
Comment (igomez06) [inline tests/v2/integration/catalogv2/rancher_managed_charts_test.go:9]: This whole file needs to be removed, this is only to be ran in v2.8 and v2.7

File: tests/v2/integration/catalogv2/cluster_repo_test.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ const (
 )
 
 var (
-	PollInterval = time.Duration(500 * time.Millisecond)
+	PollInterval = time.Duration(100 * time.Millisecond)
```
Comment (igomez06) [inline tests/v2/integration/catalogv2/cluster_repo_test.go:30]: remove this commit as well it's not needed for this branch

**Review by briandowns — DISMISSED** [review]

File: scripts/package [hunk 3]
```diff
@@ -21,7 +21,17 @@ if [ ${ARCH} == s390x ]; then
...
+    --build-arg VERSION=${TAG} \
+    --build-arg ETCD_UNSUPPORTED_ARCH=${ETCD_UNSUPPORTED_ARCH} \
+    --build-arg ARCH=${ARCH} \
+    --build-arg IMAGE_REPO=${REPO} \
+    --build-arg SYSTEM_CHART_DEFAULT_BRANCH=${SYSTEM_CHART_DEFAULT_BRANCH} \
+    --build-arg CHART_DEFAULT_BRANCH=${CHART_DEFAULT_BRANCH} \
+    --build-arg CATTLE_RANCHER_WEBHOOK_VERSION="${CATTLE_RANCHER_WEBHOOK_VERSION}" \
+    --build-arg CATTLE_CSP_ADAPTER_MIN_VERSION="${CATTLE_CSP_ADAPTER_MIN_VERSION}" \
+    --build-arg CATTLE_FLEET_VERSION="${CATTLE_FLEET_VERSION}" \
+    -t ${IMAGE} . --label org.opencontainers.image.created=2024-07-02T07:10:56Z --label org.opencontainers.image.revision=d7d0d9d854247d61f4018728e9a414cecf4fbb0e --label org.opencontainers.image.source=https://github.com/rancher/rancher.git --label org.opencontainers.image.url=https://github.com/rancher/rancher
```
Comment (aruiz14) [inline scripts/package:34]: This one should be cleaned too, like https://github.com/rancher/rancher/pull/46038/commits/f77d1e7689f0ad9d85c2b797ce3d0c3115e17881, right?

Comment (rafaelbreno) [reply scripts/package]: Yep, removed it!

**Review by briandowns — APPROVED** [review]

**Review by igomez06 — APPROVED** [review]
LGTM

**Pull request merged**